sky/cloud_stores.py

```diff
@@ -219,8 +220,76 @@ def get_storage_from_path(url: str) -> CloudStorage:
     return _REGISTRY[result.scheme]
 
 
+class IBMCosCloudStorage(CloudStorage):
+    """IBM Cloud Storage."""
+    # install rclone if package isn't already installed
+    # pylint: disable=line-too-long
+    _GET_RCLONE = [
+        'rclone version >/dev/null 2>&1 || curl https://rclone.org/install.sh | sudo bash',
+    ]
+
+    def is_directory(self, url: str) -> bool:
+        """Returns whether IBM COS bucket's 'url' is a directory.
+
+        In cloud object stores, a "directory" refers to a regular object whose
+        name is a prefix of other objects.
+        """
+
+        region = url.split('//')[1].split('/')[0]
+        s3 = ibm.get_cos_resource(region)
+        bucket_name, path, _ = data_utils.split_cos_path(url)
```

Comment:
```suggestion
        bucket_name, path, region = data_utils.split_cos_path(url)
        s3 = ibm.get_cos_resource(region)
```
I believe `split_cos_path` returns region as well

Reply:
Yup, you are right.